src/services/pcn-packetcapture/src/cbpfTOc.cpp

```diff
@@ -0,0 +1,490 @@
+#include <iostream>
+#include "cbpfTOc.h"
+#include "linux/bpf.h"
+#include <arpa/inet.h>
+#include <linux/filter.h>
+#include <map>
+#include <cstring>
```

Comment:
Somewhere in your code (likely in the control plane, where you get the C equivalent of your filter), please add also a debug possibility where the C code is dumped. This helps debugging this service in case something goes wrong with your compiler.

Reply:
I've added `logger()->debug("FilterCode:\n{0}", filterCode);`